pkgs/build-support/php/hooks/composer-install-hook.sh

```diff
@@ -0,0 +1,77 @@
+declare -a composerVendorCache
+declare -a installComposerDevDependencies
+declare -a runComposerScripts
+declare -a runComposerPlugins
+declare -a dontComposerInstallConfigureHook
+declare -a dontComposerInstallBuildHook
+declare -a dontComposerInstallInstallHook
+declare -a version
+
+if [[ -z "${dontComposerInstallConfigureHook-}" ]]; then
+  preConfigureHooks+=(composerInstallConfigureHook)
+fi
+
+if [[ -z "${dontComposerInstallBuildHook-}" ]]; then
+  preBuildHooks+=(composerInstallBuildHook)
+fi
+
+if [[ -z "${dontComposerInstallInstallHook-}" ]]; then
+  preInstallHooks+=(composerInstallInstallHook)
+fi
+
+composerInstallConfigureHook() {
+    echo "Executing composerInstallConfigureHook"
+
+    if [[ ! -e "${composerVendorCache}" ]]; then
+        echo "No composer cache found."
+        exit 1
+    fi
+
+    cp -fr ${composerVendorCache}/. .
+    chmod +w .composer -R
+
+    echo "Finished composerInstallConfigureHook"
+}
+
+composerInstallBuildHook() {
+    echo "Executing composerInstallBuildHook"
+
+    argstr=("--no-interaction" "--optimize-autoloader")
```

Comment:
It might be interesting to set `--classmap-authoritative` instead of `--optimize-autoloader` to squeeze as much as possible of performances.

Reply:
I [see](https://getcomposer.org/doc/articles/autoloader-optimization.md) that autoloader optimization can be enabled in `composer.json`. Therefore, I removed the `--optimize-autoloader`  so it's up to the project to add it or not.

However, I force `apcu-autoloader` to off by creating a composer override.